### ISSUE TITLE
Make accessing and mutating `parent` part of the API

### ISSUE THREAD

**Issue by rofinn** [JuliaDiffEq/ArrayInterface.jl#19]
It'd be nice if there was a common API for accessing and mutating `parent` data for wrapper array types.

**Comment by ChrisRackauckas** [comment]
Yes, that is a very good idea.

```julia
iswrapped(A)
getparent(A)
```

?

**Comment by ChrisRackauckas** [comment]
should `getparent` also be applied to wrapped number types, like Dual numbers and TrackedReal?

**Comment by rofinn** [comment]
Yeah, I think that would make sense... assuming they're using the existing type hierarchy (e.g., `TrackedReal <: Real`).

**Comment by Tokazama** [comment]
We have `parent_type` now. You can do `parent_type(T) <: T` to confirm that `T` isn't a wrapper. Does that resolve this issue?
